services/auth.js

```diff
@@ -0,0 +1,35 @@
+'use strict'
+
+const { BaseService } = require('.')
+
+function requiredAuth(serviceInstance, userKey, passKey) {
+  if (!(serviceInstance instanceof BaseService)) {
+    throw Error('serviceInstance should be an instance of BaseService')
+  }
+
+  return {
+    user: userKey ? serviceInstance.getSecret(userKey) : '',
+    pass: passKey ? serviceInstance.getSecret(passKey) : '',
+  }
+}
+
+function optionalAuth(serviceInstance, userKey, passKey) {
```

Comment:
Could we use named params/object here, or do we want consumers to explicitly declare value for the `userKey` arg when they don't need (like in the case of Azure DevOps)?

Reply:
Nice idea! Updated.